rubin_sim/scheduler/basis_functions/feasibility_funcs.py

```diff
@@ -61,6 +61,10 @@ class SunHighLimitBasisFunction(BaseBasisFunction):
     ----------
     sun_alt_limit : `float`
         The sun altitude limit (degrees). Sun must be higher than this at sunset to execute
+    time_to_12deg : `float`
+        How much time must be remaining before 12 degree twilight in the morning (minutes)
+    time_remaining : `float`
+        Minimum about of time that must be available before trying to execute (minutes)
```

Comment:
which suggests you might go for something sort of like what I tested here - https://github.com/rhiannonlynne/notebooks/blob/main/twilight_time.ipynb

Reply:
There's an asymmetry in picking when/how to start the inner solar system observations, I'm not sure what the best way to deal with it is. At sunset, we clearly want to start the observations as soon as possible (but we can't start at too high an airmass because we need to be able to complete multiple visits before points set below the alt limit). At sunrise, we can go right to the alt limit since things are rising, so that means we want to start a sequence as late as possible. 

It's all about how much margin for error we want to have, and it gets complicated because on one side that error is easy to express as an altitude-buffer and on the other side as a time-buffer. 

This was the basis function used in v3.2. We can check if yours works better and put that in for v3.3.

Reply:
Yes .. tricky. 
I assumed that the minimum time was being applied to sunrise because you ran out of time as the sun rose. Your description sounds like the minimum time could also be applied to sunset? 
This is just a True/False for the whole sky, right? Does it make sense to also have a function which is weighting towards which airmass is chosen (I guess I had thought that would come from other places, like the footprint).

Reply:
One thing with the mix of sun altitude (for sunset) and time (for sunrise) is that it's not super easy to see how those differences are working out, as these are completely different kinds of "units", so it seems harder to maintain and interpret.
Agree that maybe the way to go is to just test both.

Reply:
Updated the notebook to actually run the SunAltitudeFeasibilityFunction as well, and I think these actually behave pretty similarly, if configured in similar sorts of ways. So maybe it just depends which one looks easier to keep. 

<img width="749" alt="Screenshot 2023-09-08 at 1 57 19 PM" src="https://github.com/lsst/rubin_sim/assets/1855711/9112f483-3d1b-4b84-a725-6145f388182e">

Reply:
Having the fraction of twilight time as a kwarg might not be best. Since the amount of twilight time changes through the year sometimes it can start before it's ideal and/or not have enough time to complete a quad.  There's also the potential to set it so it will silently fail to execute (e.g., `max_fraction=0.1` and `min_time=15.0`).

Reply:
The time required to do a quad isn't really spelled out (probably should be documented somewhere or demonstrated how to calculate). The sun alt could be set to a point where quads aren't possible couldn't it?